samples/apps/logging/logging.cpp

```diff
@@ -175,6 +178,37 @@ namespace loggingapp
       return std::nullopt;
     }
 
+    static std::optional<std::string> get_scope(auto& ctx)
+    {
+      const auto parsed_query =
+        http::parse_query(ctx.rpc_ctx->get_request_query());
+      std::string error_string;
+      return http::get_query_value_opt<std::string>(
+        parsed_query, "scope", error_string);
+    }
+
+    static std::string private_records(auto& ctx)
+    {
+      return private_records(get_scope(ctx));
+    }
+
+    static std::string public_records(auto& ctx)
+    {
+      return public_records(get_scope(ctx));
+    }
+
+    static std::string private_records(const std::optional<std::string>& scope)
+    {
+      return scope ? std::string(PRIVATE_RECORDS) + "-" + *scope :
+                     PRIVATE_RECORDS;
```

Comment:
```suggestion
      return scope.has_value() ? fmt::format("{}-{}", PRIVATE_RECORDS, *scope) :
                     PRIVATE_RECORDS;
```

and similar elsewhere. `has_value()` avoids errors if we remove the `optional` wrapper in future (becomes a compile error we need to think about, rather than silently using another conversion to bool). `fmt` construction is just to be consistent with most of our other string formatting.